PSReadLine/OnImportAndRemove.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.IO;
+using System.Management.Automation;
+using System.Reflection;
+
+namespace Microsoft.PowerShell.PSReadLine
+{
+    public class OnModuleImportAndRemove : IModuleAssemblyInitializer, IModuleAssemblyCleanup
+    {
+        public void OnImport()
+        {
+            AppDomain.CurrentDomain.AssemblyResolve += ResolveAssembly;
+        }
+
+        public void OnRemove(PSModuleInfo module)
+        {
+            AppDomain.CurrentDomain.AssemblyResolve -= ResolveAssembly;
+        }
+
+        /// <summary>
+        /// Load the correct 'Polyfiller' assembly based on the runtime.
+        /// </summary>
+        private static Assembly ResolveAssembly(object sender, ResolveEventArgs args)
+        {
+            if (args.Name != "Microsoft.PowerShell.PSReadLine.Polyfiller, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null")
+            {
+                return null;
+            }
+
+            string root = Path.GetDirectoryName(typeof(OnModuleImportAndRemove).Assembly.Location);
+            string subd = (Environment.Version.Major >= 5) ? "net5.0" : "net461";
+            string path = Path.Combine(root, subd, "Microsoft.PowerShell.PSReadLine.Polyfiller.dll");
+
+            return Assembly.LoadFrom(path);
```

Comment:
It seems it is one time operation. Can we exclude the event handler and load the dll in an init code at startup?

Reply:
Registering the `AssemblyResolve` event make it able to load the assembly lazily, which is especially desired for a module that gets loaded at startup.

Reply:
It is critical too. People complain that starting a pwsh session is way slower than Windows PowerShell, and that's for interactive use. Also, the predictive intellisense feature is off by default.